pkg/deviceshifu/deviceshifumqtt/deviceshifumqtt_test.go

```diff
@@ -25,6 +38,41 @@ func TestMain(m *testing.M) {
 	}
 }
 
+func TestNewMQTT(t *testing.T) {
+	testCases := []struct {
+		Name      string
+		metaData  *deviceshifubase.DeviceShifuMetaData
+		expErrStr string
+		fn 		  func()
+	}{
+		{
+			"case 1 deviceshifubase.New err",
+			&deviceshifubase.DeviceShifuMetaData{
+				Name: "test",
+				ConfigFilePath: "etc/edgedevice/config",
+				Namespace:"default",
+			},
+			"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined",
+			func() {
+			},
+		},
+		//TODO : TestNew KubeConfigPath mock k8s API
+	}
+	for _, c := range testCases {
+		t.Run(c.Name, func(t *testing.T) {
+			c.fn()
+			deviceShifu, err := New(c.metaData)
+			if len(c.expErrStr) > 0 {
+				assert.Equal(t, c.expErrStr, err.Error())
+				assert.Nil(t, deviceShifu)
+			} else {
+				assert.Equal(t, c.expErrStr,"")
```

Comment:
i think this assert is not necessary for in this case len(c.expErrStr) == 0 that assert always return true